pir/cpp/client.cpp

```diff
@@ -60,14 +60,65 @@ StatusOr<uint64_t> InvertMod(uint64_t m, const seal::Modulus& mod) {
   return inverse;
 }
 
-StatusOr<Request> PIRClient::CreateRequest(std::size_t desired_index) const {
+StatusOr<Request> PIRClient::CreateRequest(
+    const std::vector<std::size_t>& indexes) const {
   const auto poly_modulus_degree =
       context_->EncryptionParams().poly_modulus_degree();
-  if (desired_index >= context_->Params()->database_size()) {
-    return InvalidArgumentError("invalid index");
+
+  vector<vector<Ciphertext>> queries(indexes.size());
+
+  for (size_t idx = 0; idx < indexes.size(); ++idx) {
+    RETURN_IF_ERROR(createQueryFor(indexes[idx], queries[idx]));
+  }
+
+  GaloisKeys gal_keys;
+  RelinKeys relin_keys;
+  try {
+    gal_keys =
+        keygen_->galois_keys_local(generate_galois_elts(poly_modulus_degree));
+    relin_keys = keygen_->relin_keys_local();
+  } catch (const std::exception& e) {
+    return InternalError(e.what());
+  }
+
+  Request request_proto;
+  RETURN_IF_ERROR(SaveRequest(queries, gal_keys, relin_keys, &request_proto));
+
+  return request_proto;
+}
+
+StatusOr<std::vector<int64_t>> PIRClient::ProcessResponse(
```

Comment:
Suggestion: If you move ProcessResponse to after createQueryFor, it'll make the diff cleaner and make it easier to follow the code.